src/IssueViz.Security/Hotspots/HotspotsList/ViewModels/HotspotsControlViewModel.cs

```diff
@@ -111,10 +109,7 @@ public async System.Threading.Tasks.Task UpdateHotspotsListAsync()
         /// Allow the observable collection <see cref="Hotspots"/> to be modified from non-UI thread. 
         /// </summary>
         private void AllowMultiThreadedAccessToHotspotsList()
-        {
-            threadHandling.ThrowIfNotOnUIThread();
-            BindingOperations.EnableCollectionSynchronization(Hotspots, Lock);
-        }
+            => threadHandling.RunOnUIThread(() => BindingOperations.EnableCollectionSynchronization(Hotspots, Lock));
```

Comment:
As discussed offline, we decided to change the code so that it doesn't matter what thread it is called on; this class will do the right thing.